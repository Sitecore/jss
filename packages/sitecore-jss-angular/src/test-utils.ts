/**
 * Remove angular comments and angular-specific bindings
 * @param {string} html
 */
export function cleanHtml(html: string): string {
  return html
    .replace(/<!--[^>]*-->/g, '')
    .replace(/\s*ng-reflect-[^=]*="[^"]*"/g, '')
    .trim();
}
