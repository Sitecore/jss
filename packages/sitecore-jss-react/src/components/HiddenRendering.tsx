import React from 'react';

const styles = {
  backgroundImage:
    'linear-gradient(45deg, #ffffff 25%, #dcdcdc 25%, #dcdcdc 50%, #ffffff 50%, #ffffff 75%, #dcdcdc 75%, #dcdcdc 100%)',
  backgroundSize: '3px 3px',
  display: 'flex',
  justifyContent: 'center',
  alignItems: 'center',
  padding: '30px',
  color: '#aaa',
};

export const HiddenRendering = () => <div style={styles}>The component is hidden</div>;
