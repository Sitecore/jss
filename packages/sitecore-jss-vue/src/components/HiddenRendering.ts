import { h, defineComponent } from 'vue';

export const HiddenRendering = defineComponent({
  render() {
    return h('div', {
      style: {
        height: '100px',
        backgroundSize: '3px 3px',
        backgroundImage:
          'linear-gradient(45deg, #ffffff 25%, #dcdcdc 25%, #dcdcdc 50%, #ffffff 50%, #ffffff 75%, #dcdcdc 75%, #dcdcdc 100%)',
      },
    });
  },
});

export const HIDDEN_RENDERING_NAME = 'Hidden Rendering';
