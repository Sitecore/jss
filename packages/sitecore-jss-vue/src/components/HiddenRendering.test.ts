import { mount } from '@vue/test-utils';
import { HiddenRendering } from './HiddenRendering';

describe('<HiddenRendering />', () => {
  it('should render', () => {
    const renderedComponent = mount(HiddenRendering);
    expect(renderedComponent.html()).toMatchSnapshot();
  });
});
