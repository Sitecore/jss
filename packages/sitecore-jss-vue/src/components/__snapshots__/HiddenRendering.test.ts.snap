// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HiddenRendering /> should render 1`] = `
<div style="height: 100px; background-size: 3px 3px;">
</div>
`;
